srcpkgs/cjs/template

```diff
@@ -1,27 +1,29 @@
 # Template file for 'cjs'
 pkgname=cjs
-version=4.6.0
+version=4.8.0
 revision=1
-build_style=gnu-configure
+build_style=meson
 build_helper=gir
-configure_args="--disable-static --disable-profiler"
-hostmakedepends="automake autoconf-archive libtool gnome-common pkg-config
- gobject-introspection glib-devel dbus mozjs52-devel"
+configure_args="-Dskip_dbus_tests=true -Dskip_gtk_tests=true
+ -Dprofiler=disabled -Dinstalled_tests=false" # Disable tests because they need dbus, X and GTK, same as gjs.
+hostmakedepends="libtool gnome-common pkg-config
+ gobject-introspection glib-devel dbus mozjs78-devel"
 makedepends="dbus-glib-devel readline-devel gobject-introspection
- gtk+3-devel mozjs52 mozjs52-devel"
+ gtk+3-devel mozjs78 mozjs78-devel"
 short_desc="Javascript for Cinnamon"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="LGPL-2.0-or-later"
 homepage="http://developer.linuxmint.com/projects/cinnamon-projects.html/"
 distfiles="https://github.com/linuxmint/cjs/archive/${version}.tar.gz"
-checksum=47f8abb867881594766fec1398d1b9117012826c5e72ecafb3153a0c8226bf81
+checksum=6f07defa6b21efcc4332ba46f286ed303a05af8805d47a9ff4bfa40cdf45b40b
 
-
-pre_configure() {
-	cp ${FILESDIR}/ax_code_coverage.m4 m4/
-
-	NOCONFIGURE=1 ./autogen.sh
-}
+if [ "$CROSS_BUILD" ]; then
+	# The deleted lines in meson.build try to build and run a simple program,
+	# and that does not work when cross-compiling.
+	pre_configure() {
+		sed -i '235,254d' meson.build
```

Comment:
I think it's best to use a patch here, since any changes to the file can change the line positions.

Reply:
I am not making a patch just to remove lines, sorry.

Reply:
Either
```sed
/Check if a minimal SpiderMonkey program compiles/,/^endif/d
```

or `build_helper=qemu`

Reply:
I do not think the test makes much sense in this case, so I do not think using a build helper is the right thing to do.

Reply:
I don't care if you decide to use helper or not, but sed to delete with 2 numbers is NOT cool.